client/src/pages/login/index.js

```diff
@@ -1,33 +1,88 @@
 import React, { Component } from "react";
+import SweetAlert from "react-bootstrap-sweetalert";
 import { Redirect, withRouter } from "react-router-dom";
 
 import { Input, Button } from "../../components";
 import "./login.css";
 
 class Login extends Component {
   state = {
-    isLogin: false,
+    isLoggedIn: false,
+    alert: null,
     loginInfo: {}
   };
 
+  invalidPasswordOrUsernameAlert() {
+    const getAlert = () => (
+      <SweetAlert
+        warning
+        title="Wrong!"
+        confirmBtnBsStyle="danger"
+        s
+        cancelBtnBsStyle="danger"
+        onConfirm={() => this.hideAlert()}
+      >
+        Invalied Password or Username !
+      </SweetAlert>
+    );
+
+    this.setState({
+      alert: getAlert()
+    });
+  }
+
+  hideAlert() {
+    this.setState({
+      alert: null
+    });
+  }
+
   handleLogin = e => {
     e.preventDefault();
-    this.setState({
-      isLogin: !this.state.isLogin
+    const { username, password } = this.state.loginInfo;
+    const data = JSON.stringify({
+      username,
+      password
     });
+    fetch("/login_post", {
```

Comment:
(Copying this comment over from #26)

Here's a new set of keywords I learned recently, which you could use here (or in the future with promises):
`async` and `await` - if you declare your function with "async", you can use "await" inside it, to wait for promises without needing to write "then" functions. It's a new ES2017 thing, but you're using babel-preset-env so you are already set up in your build steps to write this style.

For example:
```
async handleLogin(e) {
  ...
  const response = await fetch('/login_post', { .... })
  //there's no then() needed!
  responseBody = response.json()
  ...
}
```